XamarinCommunityToolkit/Effects/Touch/PlatformTouchEffect.ios.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using CoreGraphics;
+using Foundation;
+using UIKit;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.CommunityToolkit.iOS.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+
+[assembly: ExportEffect(typeof(PlatformTouchEffect), nameof(TouchEffect))]
+
+namespace Xamarin.CommunityToolkit.iOS.Effects
+{
+	public class PlatformTouchEffect : PlatformEffect
+	{
+		UIGestureRecognizer gesture;
+		TouchEffect effect;
+
+		protected override void OnAttached()
+		{
+			effect = TouchEffect.PickFrom(Element);
+			if (effect?.IsDisabled ?? true)
+				return;
+
+			effect.Control = Element as VisualElement;
+
+			gesture = new TouchUITapGestureRecognizer(effect);
+			if (Container != null)
+			{
+				Container.AddGestureRecognizer(gesture);
+				Container.UserInteractionEnabled = true;
+			}
+		}
+
+		protected override void OnDetached()
+		{
+			if (effect?.Control == null)
+				return;
+
+			Container?.RemoveGestureRecognizer(gesture);
+			gesture?.Dispose();
+			gesture = null;
+			effect.Control = null;
+			effect = null;
+		}
+	}
+
+	sealed class TouchUITapGestureRecognizer : UIGestureRecognizer
+	{
+		TouchEffect effect;
```

Comment:
can we make this a `WeakReference`, the same motivation as Android listener.

Reply:
We clear this reference in the Dispose method. Are you sure we need it?